storage-proofs-post/src/fallback/utils.rs

```diff
@@ -0,0 +1,14 @@
+use storage_proofs_core::api_version::ApiVersion;
+
+/// Selects the challenge index used to determine the leaf challenge for PoSt
+pub fn get_challenge_index(
+    api_version: ApiVersion,
+    sector_index: usize,
+    challenge_count_per_sector: usize,
+    challenge_index: usize,
+) -> u64 {
+    (match api_version {
+        ApiVersion::V1_2_0 => challenge_index,
+        _ => sector_index * challenge_count_per_sector + challenge_index,
```

Comment:
I propose making this exhaustive with using all available `ApiVersion`. This way it will fail at compile time when a new one is added. In the current code, a new `ApiVersion` would silently use the non v1_2_0, although I guess a v1_3_0, should use the v1_2_0 one.

Reply:
Thanks, I took a shot at applying all review feedback!